Mathlib/Probability/Distributions/Gaussian.lean

```diff
@@ -347,6 +346,48 @@ lemma gaussianReal_mul_const {X : Ω → ℝ} (hX : Measure.map X ℙ = gaussian
 
 end Transformations
 
+open Measurable Real
+
+variable {Ω : Type*} {mΩ : MeasurableSpace Ω} {p : Measure Ω} {μ : ℝ} {v : ℝ≥0} {X : Ω → ℝ}
+
+lemma mgf_dirac (hX : p.map X = .dirac μ) (t : ℝ) : mgf X p t = exp (μ * t) := by
+  rw [← mgf_id_map, mgf, hX, integral_dirac, mul_comm]
+  · rfl
+  · apply AEMeasurable.of_map_ne_zero
+    rw [hX]
+    exact IsProbabilityMeasure.ne_zero (.dirac μ)
+
+theorem mgf_gaussianReal (hX : p.map X = gaussianReal μ v) (t : ℝ) :
+    mgf X p t = exp (μ * t + v * t ^ 2 / 2) := by
+  by_cases hv : v = 0
+  · simp only [gaussianReal, hv, ↓reduceIte] at hX
+    simp [mgf_dirac hX, hv]
+  calc
+  mgf X p t = (p.map X)[fun x => exp (t * x)] := by
+    rw [← mgf_id_map, mgf]
+    · rfl
+    · apply AEMeasurable.of_map_ne_zero
+      rw [hX]
+      exact IsProbabilityMeasure.ne_zero (gaussianReal μ v)
+  _ = ∫ x, exp (t * x) * gaussianPDFReal μ v x := by
+    simp [hX, gaussianReal_of_var_ne_zero μ hv, gaussianPDF_def, ENNReal.ofReal,
+      integral_withDensity_eq_integral_smul (measurable_gaussianPDFReal μ v).real_toNNReal,
+      NNReal.smul_def, coe_toNNReal', gaussianPDFReal_nonneg, sup_of_le_left, smul_eq_mul, mul_comm]
```

Comment:
```suggestion
      NNReal.smul_def, coe_toNNReal', gaussianPDFReal_nonneg, sup_of_le_left, mul_comm]
```
and some more lemmas might already be simp, like `coe_toNNReal', gaussianPDFReal_nonneg, sup_of_le_left` (actually, the middle one currently isn't, but should be)